### DISCUSSION TITLE
Lifecycle hooks Runs before the form fields are saved to the database. document have issue with wrong example

### DISCUSSION THREAD

**Discussion by spsarola** [filamentphp/filament#11844] [General]
I am trying to use Lifecycle hooks Runs before the form fields are saved to the database and follow the document
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#lifecycle-hooks

in this the following code snippet 
`
    CreateAction::make()
		->before(function (CreateAction $action, Post $record) {
			info("before created called");
			info($record);
		})
`

and I got the following error Argument #2 ($record) must be of type App\Models\Post, null given, called in D:\wamp64\www\filament-demo\vendor\filament\support\src\Concerns\EvaluatesClosures.php on line 35

I am also confused because before creating the model How do I get the mode in before creating the lifecycle hook.?

**Comment by indigoram89** [comment]
Hello, guys, I have the same error. Any information about it?

**Reply by leandrocfe** [reply]
I think you need to inject `array $data` instead of $record
